genie/input_to_database.py

```diff
@@ -67,40 +69,36 @@ def get_center_input_files(syn, synid, center, process="main"):
     center_files = synapseutils.walk(syn, synid)
     clinicalpair = []
     prepared_center_file_list = []
-    for dirpath, dirname, filenames in center_files:
-        for name, ent_synid in filenames:
-            logger.info(name)
-            paired = False
-            '''
-            Clinical file can come as two files.
-            The two files need to be merged together which is
-            why there is this format
-            '''
+    
+    for _, _, entities in center_files:
+        for name, ent_synid in entities:
+            # This is to remove vcfs from being validated during main 
+            # processing. Often there are too many vcf files, and it is 
+            # not necessary for them to be run everytime.
+            if name.endswith(".vcf") and process != "vcf":
+                continue
+
+            ent = syn.get(ent_synid)
+            logger.debug(ent)
+
+            # Clinical file can come as two files.
+            # The two files need to be merged together which is
+            # why there is this format
+
             if name in clinical_pair_name:
-                paired = True
-                clinicalpair.append(ent_synid)
-            if len(clinicalpair) == 2:
-                syns = [i for i in clinicalpair]
-                paths = [rename_file(syn, i) for i in clinicalpair]
-                clinicalpair = []
-                prepared_center_file_list.append((syns, paths))
-            elif not paired:
-                '''
-                This is to remove vcfs from being validated
-                during main processing.  Often there are too many vcf files.
-                Not necessary for them to be run everytime.
-                '''
-                if process == "vcf":
-                    prepared_center_file_list.append(
-                        ([ent_synid], [rename_file(syn, ent_synid)]))
-                elif not name.endswith(".vcf"):
-                    prepared_center_file_list.append(
-                        ([ent_synid], [rename_file(syn, ent_synid)]))
-    return(prepared_center_file_list)
-
-
-def check_existing_file_status(validation_statusdf, error_trackerdf,
-                               entities, input_filenames):
+                clinicalpair.append(ent)
+                continue
+
+            prepared_center_file_list.append([rename_file(syn, ent)])
+    
+    if clinicalpair:
```

Comment:
This should be if len(clinicalpair) == 2.  Because if someone uploads just a patient file, it should error out.

Reply:
This function does not make assumptions about why something is bad. it only collects the information. There needs to be a function that 'validates' (for lack of a better word) that the submission package is correct. That logic should be elsewhere.

Reply:
What COULD happen here is that instead of returning file lists, you return an instance of a `FileTypeFormat`, which internally use entities instead of file paths.